src/main/java/teammates/ui/webapi/action/GetCourseStudentDetailsAction.java

```diff
@@ -60,7 +60,10 @@ public ActionResult execute() {
             return new JsonResult("No course with given id.", HttpStatus.SC_NOT_FOUND);
         }
 
-        StudentProfileAttributes studentProfile = logic.getStudentProfile(student.googleId);
+        StudentProfileAttributes studentProfile = null;
+        if (!"".equals(student.googleId)) {
```

Comment:
You can change this condition to `student.isRegistered()`.

Reply:
Done!